modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,388 @@
+package org.tessellation.infrastructure.trust.storage
+
+import cats.kernel.Order
+import cats.syntax.option._
+import cats.syntax.partialOrder._
+import cats.syntax.semigroup._
+
+import org.tessellation.infrastructure.trust.generators.{genPeerLabel, genPeerObservationAdjustmentUpdate, genPeerTrustInfo}
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.schema.generators.peerIdGen
+import org.tessellation.schema.trust._
+import org.tessellation.sdk.domain.trust.storage.OrdinalTrustMap.{ordinalTrustUpdateDelay, ordinalTrustUpdateInterval}
+import org.tessellation.sdk.domain.trust.storage.TrustMap.isEmpty
+import org.tessellation.sdk.domain.trust.storage.{OrdinalTrustMap, PublicTrustMap, TrustMap}
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.numeric.NonNegative
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object TrustStorageSuite extends SimpleIOSuite with Checkers {
+  test("initial trust storage is empty") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getTrust
+    } yield expect.all(isEmpty(trust))
+  }
+
+  test("internal trust storage is updated with trust labels") {
+    val gen = Gen.listOfN(4, genPeerObservationAdjustmentUpdate).map(labels => PeerObservationAdjustmentUpdateBatch(labels))
+
+    forall(gen) { labels =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updateTrust(labels)
+
+        expected = labels.updates.map(update => update.id -> TrustInfo(trustLabel = update.trust.value.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          isEmpty(firstTrust),
```

Comment:
if you already checked that `initial trust storage is empty`, why you check it also in each and every other test case?

Reply:
great, these tests are needed indeed. 
```scala
  test("initial trust storage is empty") {
    for {
      store <- mkTrustStorage()
      trust <- store.getTrust
    } yield expect.all(trust.isEmpty)
  }
```

but in almost every other test case you also wrote an assertion to check if these storages are empty so you double the assertion in each test. if you already checked that, there is no need to do that again and again
```scala
 test("internal trust storage is updated with trust labels") {
    for {
        store <- mkTrustStorage()
        firstTrust <- store.getTrust

        expect.all(
          firstTrust.isEmpty,
          ...
        )
    }
  }
```

Reply:
De-duplicated the assertions.